src/graphnet/data/datamodule.py

```diff
@@ -0,0 +1,429 @@
+        print(
+            f"Type of self._dataset before validation check: {type(self._dataset)}"
+        )
+        # if type(self._dataset) not in [SQLiteDataset, ParquetDataset, Dataset]:
+        #     raise TypeError(
+        #         "dataset_reference must be an instance of SQLiteDataset, ParquetDataset, or Dataset."
+        #     )
+        if isinstance(self._dataset, EnsembleDataset):
+            raise TypeError(
+                "EnsembleDataset is not allowed as dataset_reference."
+            )
+
+    def _validate_dataset_args(self) -> None:
+        """Sanity checks on the arguments for the dataset reference."""
+        if isinstance(self._dataset_args["path"], list):
+            if self._selection is not None:
+                try:
+                    # Check that the number of dataset paths is equal to the
+                    # number of selections given as arg.
+                    assert len(self._dataset_args["path"]) == len(
+                        self._selection
+                    )
+                except AssertionError:
+                    raise ValueError(
+                        f"The number of dataset paths ({len(self._dataset_args['path'])}) does not match the number of selections ({len(self._selection)})."
+                    )
+
+            if self._test_selection is not None:
+                try:
+                    # Check that the number of dataset paths is equal to the
+                    # number of test selections.
+                    assert len(self._dataset_args["path"]) == len(
+                        self._test_selection
+                    )
+                except AssertionError:
+                    raise ValueError(
+                        f"The number of dataset paths ({len(self._dataset_args['path'])}) does not match the number of test selections ({len(self._test_selection)}). If you'd like to test on only a subset of the {len(self._dataset_args['path'])} datasets, please provide empty test selections for the others."
+                    )
+
+    def _validate_dataloader_args(self) -> None:
+        """Sanity check on `dataloader_args`."""
+        if "dataset" in self._train_dataloader_kwargs:
+            raise ValueError(
+                "`train_dataloader_kwargs` must not contain `dataset`"
+            )
+        if "dataset" in self._validation_dataloader_kwargs:
+            raise ValueError(
+                "`validation_dataloader_kwargs` must not contain `dataset`"
+            )
+        if "dataset" in self._test_dataloader_kwargs:
+            raise ValueError(
+                "`test_dataloader_kwargs` must not contain `dataset`"
+            )
+
+    def _resolve_selections(self) -> None:
+        if self._test_selection is None:
+            self.warning_once(
+                f"{self.__class__.__name__} did not receive an argument for `test_selection` and will therefore not have a prediction dataloader available."
```

Comment:
This can be shortened like the `ValueError`s. Something like: 
```python
self.warning_once(f"{self.__class__.__name__} did not receive an" 
                  " argument for `test_selection` and will "
                  "therefore not have a prediction dataloader available.")
```